### PR TITLE
Cortx 33329 data pod restart testsuite

### PR DESCRIPTION
**Pull request by sonalk0209**
# Problem Statement
-V9 changes for data pod restart test suite

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
-  [ ] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: tests/ha/single_pod_restart/test_data_pod_restart.py [hunk 1]
```diff
@@ -252,17 +267,18 @@ def test_write_after_pod_restart(self):
...
-            down_method=const.RESTORE_DEPLOYMENT_K8S)
+            delete_pod=[self.delete_pod], num_replica=num_replica)
         # Assert if empty dictionary
         assert_utils.assert_true(resp[1], "Failed to shutdown/delete pod")
         pod_name = list(resp[1].keys())[0]
-        self.deployment_name = resp[1][pod_name]['deployment_name']
-        self.deployment_backup = resp[1][pod_name]['deployment_backup']
+        self.set_name = resp[1][pod_name]['deployment_name']
         self.restore_method = resp[1][pod_name]['method']
+        pod_name = list(resp[1].keys())[0]
```
Comment (ApurwaMohite) [inline tests/ha/single_pod_restart/test_data_pod_restart.py:281]: Statefulset changes are needed for restarting pod also, please check at all the places.

**Comment by sonalk0209** [comment]
image

File: tests/ha/single_pod_failure/test_data_pod_failure.py [hunk 2]
```diff
@@ -155,7 +155,8 @@ def teardown_method(self):
                 resp = self.ha_obj.delete_s3_acc_buckets_objects(self.s3_clean)
             else:
                 resp = self.ha_obj.delete_s3_acc_buckets_objects(self.s3_clean, obj_crud=True)
-            assert_utils.assert_true(resp[0], resp[1])
+            if not resp[0]:
+                LOGGER.error("Failed to delete objects/buckets")
```
Comment (ankita-93) [inline tests/ha/single_pod_failure/test_data_pod_failure.py:159]: This needs to be removed once we do not have to re-deploy after every test, right ?

So can we add TODO, accordingly ?

File: tests/ha/single_pod_failure/test_server_pod_failure.py [hunk 3]
```diff
@@ -138,7 +138,8 @@ def teardown_method(self):
         if self.s3_clean:
             LOGGER.info("Cleanup: Cleaning created IAM users and buckets.")
             resp = self.ha_obj.delete_s3_acc_buckets_objects(self.s3_clean)
-            assert_utils.assert_true(resp[0], resp[1])
+            if not resp[0]:
+                LOGGER.error("Failed to delete objects/buckets")
```
Comment (ankita-93) [inline tests/ha/single_pod_failure/test_server_pod_failure.py:142]: same here

File: tests/ha/single_pod_restart/test_data_pod_restart.py [hunk 4]
```diff
@@ -121,6 +121,18 @@ def setup_method(self):
         resp = self.ha_obj.check_cluster_status(self.node_master_list[0])
         assert_utils.assert_true(resp[0], resp[1])
         LOGGER.info("Precondition: Verified cluster is up and running and all pods are online.")
+        LOGGER.info("Get %s pod to be deleted", const.POD_NAME_PREFIX)
```
Comment (ankita-93) [inline tests/ha/single_pod_restart/test_data_pod_restart.py:124]: Why this change ?
And if it's common, can we add lib for this, as we'll need these code in most of the files.

File: tests/ha/single_pod_failure/test_data_pod_failure.py (hunk 2 above)
Comment (sonalk0209) [reply tests/ha/single_pod_failure/test_data_pod_failure.py:159]: No the thought was that in any case, we should go for at least restoring the pod or else rest of the tests of the suite will start failing. And as such deletes will be tested in deletes test cases. What u think?

File: tests/ha/single_pod_failure/test_server_pod_failure.py (hunk 3 above)
Comment (sonalk0209) [reply tests/ha/single_pod_failure/test_server_pod_failure.py:142]: same as above

File: tests/ha/single_pod_restart/test_data_pod_restart.py (hunk 4 above)
Comment (sonalk0209) [reply tests/ha/single_pod_restart/test_data_pod_restart.py:124]: We are adding this for now in setup method untill we get another way for pod deletion. Once we have another method we can add a wrapper lib to choose a method but till then we can keep this as is.

**Review by ankita-93 — APPROVED** [review]

File: tests/ha/single_pod_restart/test_data_pod_restart.py [hunk 5]
```diff
@@ -1026,17 +1059,19 @@ def test_continuous_reads_during_pod_restart(self):
             thread1.start()
         LOGGER.info("Step 5: Successfully started READs and verified DI on the written data in "
                     "background")
-        LOGGER.info("Step 6: Starting pod again by creating deployment using K8s command")
+        LOGGER.info("Step 6: Starting pod again and checking cluster status")
```
Comment (ApurwaMohite) [inline tests/ha/single_pod_restart/test_data_pod_restart.py:1062]: We have to pass event_set_clr list to s3 bench IOs for background IOs. Or those changes we will do in separate PR?

File: tests/ha/single_pod_restart/test_data_pod_restart.py [hunk 6]
```diff
@@ -1567,17 +1609,19 @@ def test_ios_during_pod_restart(self):
         LOGGER.info("Wait for %s seconds for all background operations to start",
                     HA_CFG["common_params"]["30sec_delay"])
         time.sleep(HA_CFG["common_params"]["30sec_delay"])
-        LOGGER.info("Step 7: Starting pod again by creating deployment using K8s command")
+        LOGGER.info("Step 7: Starting pod again and checking cluster status")
```
Comment (ApurwaMohite) [inline tests/ha/single_pod_restart/test_data_pod_restart.py:1612]: Same as above, event_set_clr list... we will have to check this for all s3 bench background IOs.

File: tests/ha/single_pod_restart/test_data_pod_restart.py (hunk 5 above)
Comment (sonalk0209) [reply tests/ha/single_pod_restart/test_data_pod_restart.py:1062]: As we are expecting pass for all IO, I don't think we need to set it.

File: tests/ha/single_pod_restart/test_data_pod_restart.py (hunk 6 above)
Comment (sonalk0209) [reply tests/ha/single_pod_restart/test_data_pod_restart.py:1612]: same reply as above

**Review by ApurwaMohite — APPROVED** [review]

**Review by ravindrachoudhari — APPROVED** [review]
